tests/tracer/telemetry/test_writer.py

```diff
@@ -0,0 +1,243 @@
+import httpretty
+import mock
+import pkg_resources
+import pytest
+
+from ddtrace.internal.telemetry.data import get_application
+from ddtrace.internal.telemetry.data import get_host_info
+from ddtrace.internal.telemetry.writer import TelemetryWriter
+from ddtrace.internal.telemetry.writer import get_runtime_id
+from ddtrace.settings import _config as config
+
+
+AGENT_URL = "http://localhost:8126"
+TEST_URL = "%s/%s" % (AGENT_URL, TelemetryWriter.ENDPOINT)
+
+
+@pytest.fixture
+def mock_status():
+    return 202
+
+
+@pytest.fixture
+def mock_send_request(mock_status):
+    with httpretty.enabled():
+        httpretty.register_uri(httpretty.POST, TEST_URL, status=mock_status)
+        yield
```

Comment:
It might be more natural to make this a context manager instead of hacking around pytest fixtures.

```python
@contextlib.contextmanager
def mock_send_request(status=202):
    with httpretty.enabled():
        httpretty.register_uri(httpretty.POST, TEST_URL, status=status)
        yield


def test_thing():
    with mock_send_request():
        # test things
```

what do you think? (there are probably other options to address)

Reply:
ignore this, this pattern is new to me, but recommended way to do it in pytest, neat!